providers/image-service.ts

```diff
@@ -49,4 +62,91 @@ export class ImageService {
   getSanitizedImageBackground(url: string) {
     return this.sanitizer.bypassSecurityTrustStyle(`url(${url})`);
   }
+
+  /**
+   * Get image from the camera and just return the image data without uploading.
+   * @param direction The direction for the camera (`front` or `back`)
+   */
+  getImageFromCamera(direction?: string, width?: number, height?: number){
+    return this.loaderService.showLoader()
+    .switchMap(obs => Rx.Observable.of(this.app.isMobilePlatform()))
+    .flatMap(isMobile => {
+        this.loaderService.dismissLoader(); // Dismiss loader before opening Camera activity to avoid losing reference
+
+        if(isMobile) {
+            const options: CameraOptions = {
+                quality: 25,
+                destinationType: this.camera.DestinationType.DATA_URL,
+                encodingType: this.camera.EncodingType.JPEG,
+                mediaType: this.camera.MediaType.PICTURE,
+                correctOrientation: true
+              };
+
+              if(direction && direction === 'front') options['cameraDirection'] = this.camera.Direction.FRONT;
+              if(width && width > 0) options['targetWidth'] = width;
+              if(height && height > 0) options['targetHeight'] = height;
+
+
+              return Rx.Observable.fromPromise(this.camera.getPicture(options));
+        } else {
+            // Return mocked image
+            return Rx.Observable.of(this.mockService.mockBase64Image());
+        }
+    })
+    .flatMap((imageData: any) => {
+        return Rx.Observable.of(imageData); //Just return the image data for further use.
+    });
+  }
+
+  getImageFromCameraAndUpload() {
+      return this.loaderService.showLoader()
+      .switchMap(obs => Rx.Observable.of(this.app.isMobilePlatform()))
+      .flatMap(isMobile => {
+          this.loaderService.dismissLoader(); // Dismiss loader before opening Camera activity to avoid losing reference
+
+          if(isMobile) {
+              const options: CameraOptions = {
+                  quality: 25,
+                  destinationType: this.camera.DestinationType.DATA_URL,
+                  encodingType: this.camera.EncodingType.JPEG,
+                  mediaType: this.camera.MediaType.PICTURE,
+                  correctOrientation: true
+                };
+
+                return Rx.Observable.fromPromise(this.camera.getPicture(options));
+          } else {
+              // Return mocked image
+              return Rx.Observable.of(this.mockService.mockBase64Image());
+          }
+      })
+      .flatMap((imageData: any) => {
+          return this.loaderService.showLoader() // Show loader again
+              .switchMap(obs => this.updloadToCdn(imageData));
+      })
+      .flatMap((cdnResponse: any) => {
+          return Rx.Observable.of(cdnResponse.cdnUrl);
+      });
+  }
+
+  updloadToCdn(imageData: any) {
+      return Rx.Observable.create(observer => { // Extract picture
+          // 'data:image/jpeg;base64,' +
```

Comment:
this line is reported below, pls remove the comment

Reply:
Perfect. Removed the comment 🙃